src/Gzero/Core/Query/Condition.php

```diff
@@ -181,20 +187,34 @@ protected function validate()
         if (!in_array($this->operation, self::$allowedOperations, true)) {
             throw new InvalidArgumentException('Unsupported condition operation');
         }
-        if (is_array($this->value) && !$this->isCorrectRangeFormat()) {
+
+        if (!is_array($this->value) && $this->isArrayOperation()) {
+            throw new InvalidArgumentException('Value is not of type array');
+        }
+
+        if (is_array($this->value) && $this->isDateOperation() && count($this->value) < 2) {
             throw new InvalidArgumentException('Wrong number of values for range');
         }
     }
 
     /**
-     * Checks if it's correct range format
+     * Check if operation is an array operation
+     *
+     * @return bool
+     */
+    protected function isArrayOperation(): bool
+    {
+        return $this->operation === 'in' || $this->operation === 'not in';
+    }
+
+    /**
+     * Check if operation is a date operation
      *
      * @return bool
      */
-    protected function isCorrectRangeFormat(): bool
+    protected function isDateOperation(): bool
```

Comment:
I would stick with the old nomenclature. I think it's better. SQL's `between` can be used for any type of data, which can be ordered NOT just dates.
Let's rather name it then: `isRangeOperation`